src/components/FormBuilder/FormBuilder.js

```diff
@@ -0,0 +1,204 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import './FormBuilder.scss';
+import SectionContainer from '../SectionContainer';
+import GroupContainer from '../GroupContainer';
+import InputFieldsRenderer from '../InputFieldsRenderer';
+import { getDefaultParamsForFormElement } from '../../constants';
+
+/*
+1. All elements have id as - section_12324-group_345345-formElement_32343534
+2. We need the user to create sections. They can't directly put in form elements.
+3. Once user has created sections, they can either create groups or directly put in form elements. Fobu supports both.
+4. Schema looks like below -
+[
+  id: section_12324,
+  type: 'section',
+  title: "Section Title",
+  description: 'Section description',
+  children: [
+    {
+      id: section_12324-group_345345,
+      type: 'group',
+      title: "Group Title",
+      description: 'Group description',
+      children: [
+        {
+          id: section_12324-group_345345-formElement_32343534,
+          type: "input",
+          elementParams: {type: "text", label: "Your label here"}
+        }
+      ]
+    }
+  ]
+]
+*/
+
+class FormBuilder extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      formSchema: [],
+      formData: {},
+      settingsModalOpen: false
+    };
+  }
+
+  generateId(nodeType, parentId) {
+    return `${parentId ? parentId + '-' : ''}${nodeType}_${Date.now()}`;
+  }
+
+  splitFormElementId(id) {
+    let idArr = id.split('-'),
+      returnObj = {};
+    if (idArr.length == 3) {
```

Comment:
should we rely on `type` or `id` of the element to compute whether the element is a section, group or form-element? I don't think we should rely on id-path length.